examples/nlp/duplex_text_normalization/duplex_text_normalization_test.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This script contains an example on how to evaluate a DuplexTextNormalizationModel.
+Note that DuplexTextNormalizationModel is essentially a wrapper class around
+DuplexTaggerModel and DuplexDecoderModel. Therefore, two trained NeMo models
+should be specificied before evaluation (one is a trained DuplexTaggerModel
+and the other is a trained DuplexDecoderModel).
+
+USAGE Example:
+1. Obtain a processed test data file (refer to the `text_normalization doc <https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization.rst>`)
+2.
+# python duplex_text_normalization_test.py
+        tagger_pretrained_model=PATH_TO_TRAINED_TAGGER
+        decoder_pretrained_model=PATH_TO_TRAINED_DECODER
+        data.test_ds.data_path=PATH_TO_TEST_FILE
+        mode={tn,itn,joint}
+
+The script also supports the `interactive` mode where a user can just make the model
+run on any input text:
+# python duplex_text_normalization_test.py
+        tagger_pretrained_model=PATH_TO_TRAINED_TAGGER
+        decoder_pretrained_model=PATH_TO_TRAINED_DECODER
+        mode={tn,itn,joint}
+        inference.interactive=true
+
+This script uses the `/examples/nlp/duplex_text_normalization/conf/duplex_tn_config.yaml`
+config file by default. The other option is to set another config file via command
+line arguments by `--config-name=CONFIG_FILE_PATH'.
+
+"""
+
+
+import numpy as np
+import nemo.collections.nlp.data.text_normalization.constants as constants
+
+from tqdm import tqdm
+from math import ceil
+from nltk import word_tokenize
+from time import perf_counter
+from omegaconf import DictConfig, OmegaConf
+from utils import TAGGER_MODEL, DECODER_MODEL, instantiate_model_and_trainer
+
+from nemo.utils import logging
+from nemo.core.config import hydra_runner
+from nemo.collections.nlp.models import DuplexTextNormalizationModel
+from nemo.collections.nlp.data.text_normalization import TextNormalizationTestDataset
+
+@hydra_runner(config_path="conf", config_name="duplex_tn_config")
+def main(cfg: DictConfig) -> None:
+    logging.info(f'Config Params: {OmegaConf.to_yaml(cfg)}')
+    tagger_trainer, tagger_model = instantiate_model_and_trainer(cfg, TAGGER_MODEL, False)
+    decoder_trainer, decoder_model = instantiate_model_and_trainer(cfg, DECODER_MODEL, False)
```

Comment:
Have you checked eval numbers on single vs multi-GPU?

Reply:
Yes. I just tested with multi-GPU. It still works.